pandas/core/reshape/pivot.py

```diff
@@ -683,3 +697,35 @@ def _get_names(arrs, names, prefix="row"):
             names = list(names)
 
     return names
+
+
+def _build_names_mapper(names, shared_col_row_names, suffix):
```

Comment:
can you give this a docstring and possibly type annotations (only if theyre easy)

Reply:
Added docstrings and type annotations

Reply:
Added types and docstrings